tests.js

```diff
@@ -705,22 +708,55 @@ test("SASL SCRAM-SHA-1 Auth", (assert) => {
     assert.ok(saslsha1.test(conn), "SHA-1 is enabled by default.");
     // test taken from example section on:
     // URL: http://tools.ietf.org/html/rfc5802#section-5
-    let response = saslsha1.clientChallenge(conn, "fyko+d2lbbFgONRv9qkxdawL");
+    let response = await saslsha1.clientChallenge(conn, "fyko+d2lbbFgONRv9qkxdawL");
     assert.equal(response, "n,,n=user,r=fyko+d2lbbFgONRv9qkxdawL", "checking first auth challenge");
 
-    response = saslsha1.onChallenge(conn, "r=fyko+d2lbbFgONRv9qkxdawL3rfcNHYJY1ZVvWVs7j,s=QSXCR+Q6sek8bf92,i=4096");
+    response = await saslsha1.onChallenge(conn, "r=fyko+d2lbbFgONRv9qkxdawL3rfcNHYJY1ZVvWVs7j,s=QSXCR+Q6sek8bf92,i=4096");
     assert.equal(response, "c=biws,r=fyko+d2lbbFgONRv9qkxdawL3rfcNHYJY1ZVvWVs7j,p=v0X8v3Bz2T0CJGbJQyF0X+HI4Ts=",
         "checking second auth challenge");
     saslsha1.onSuccess();
 });
 
-test("SASL EXTERNAL Auth", (assert) => {
+test("SASL SCRAM-SHA-256 Auth", async (assert) => {
+    /* This is a simple example of a SCRAM-SHA-1 authentication exchange
+    * when the client doesn't support channel bindings (username 'user' and
+    * password 'pencil' are used):
+    *
+    * C: n,,n=user,r=rOprNGfwEbeRWgbNEkqO
+    * S: r=rOprNGfwEbeRWgbNEkqO%hvYDpWUa2RaTCAfuxFIlj)hNlF$k0,
+    * s=W22ZaJ0SNY7soEsUEjb6gQ==,i=4096
+    * C: c=biws,r=rOprNGfwEbeRWgbNEkqO%hvYDpWUa2RaTCAfuxFIlj)hNlF$k0,
+    * p=dHzbZapWIk4jUhN+Ute9ytag9zjfMHgsqmmiz7AndVQ=
+    * S: v=6rriTRBi23WpRR/wtup+mMhUZUn/dB5nLTJRsjl95G4=
+    *
+    */
+    const conn = {
+        pass: "pencil",
+        authcid: "user",
+        authzid: "user@xmpp.org",
+        _sasl_data: []
+    };
+    const saslsha256 = new Strophe.SASLSHA256();
+    saslsha256.onStart(conn);
+    assert.ok(saslsha256.test(conn), "SHA-256 is enabled by default.");
+    // test taken from example section on:
+    // URL: http://tools.ietf.org/html/rfc5802#section-5
+    let response = await saslsha256.clientChallenge(conn, "rOprNGfwEbeRWgbNEkqO");
+    assert.equal(response, "n,,n=user,r=rOprNGfwEbeRWgbNEkqO", "checking first auth challenge");
+
+    response = await saslsha256.onChallenge(conn, "r=rOprNGfwEbeRWgbNEkqO%hvYDpWUa2RaTCAfuxFIlj)hNlF$k0,s=W22ZaJ0SNY7soEsUEjb6gQ==,i=4096");
+    assert.equal(response, "c=biws,r=rOprNGfwEbeRWgbNEkqO%hvYDpWUa2RaTCAfuxFIlj)hNlF$k0,p=dHzbZapWIk4jUhN+Ute9ytag9zjfMHgsqmmiz7AndVQ=",
+        "checking second auth challenge");
+    saslsha256.onSuccess();
+});
+
+test("SASL EXTERNAL Auth", async (assert) => {
```

Comment:
Thanks for adding a test. Could you also please add one for SHA-512?

Reply:
Nevermind, I saw this: https://github.com/strophe/strophejs/issues/314#issuecomment-1172700269